src/executor/hash_join.rs

```diff
@@ -106,17 +98,13 @@ impl HashJoinExecutor {
                 // append row: (NULL, right)
                 let values =
                     (self.left_types.iter().map(|_| DataValue::Null)).chain(right_row.values());
-                builder = match builder.push_row(values) {
-                    Err(chunk) => {
-                        yield chunk;
-                        DataChunkBuilder::new(data_types(), PROCESSING_WINDOW_SIZE)
-                    }
-                    Ok(builder) => builder,
+                if let Some(chunk) = builder.push_row(values) {
+                    yield chunk;
                 }
             }
         }
 
-        if let Some(chunk) = builder.finish() {
+        if let Some(chunk) = { builder }.take() {
```

Comment:
Is this `{}` necessary?

Reply:
Not necessary, but consume it explicitly is also reasonable.